crates/ruff_linter/src/rules/ruff/rules/post_init_default.rs

```diff
@@ -0,0 +1,187 @@
+        default,
+        range: _,
+    } in function_def.parameters.iter_non_variadic_params()
+    {
+        let Some(default) = default else {
+            continue;
+        };
+        let mut diagnostic = Diagnostic::new(PostInitDefault, default.range());
+
+        if !stopped_fixes {
+            diagnostic.try_set_fix(|| {
+                use_initvar(current_scope, function_def, parameter, default, checker)
+            });
+            // Need to stop fixes as soon as there is a parameter we cannot fix.
+            // Otherwise, we risk a syntax error (a parameter without a default
+            // following parameter with a default).
+            stopped_fixes |= diagnostic.fix.is_none();
+        }
+
+        diagnostics.push(diagnostic);
+    }
+
+    checker.diagnostics.extend(diagnostics);
+}
+
+/// Generate a [`Fix`] to transform a `__post_init__` default argument into a
+/// `dataclasses.InitVar` pseudo-field.
+fn use_initvar(
+    current_scope: &Scope,
+    post_init_def: &ast::StmtFunctionDef,
+    parameter: &ast::Parameter,
+    default: &ast::Expr,
+    checker: &Checker,
+) -> anyhow::Result<Fix> {
+    if current_scope.has(&parameter.name) {
+        return Err(anyhow::anyhow!(
+            "Cannot add a `{}: InitVar` field to the class body, as a field by that name already exists",
+            parameter.name
+        ));
+    }
+
+    // Ensure that `dataclasses.InitVar` is accessible. For example,
+    // + `from dataclasses import InitVar`
+    let (import_edit, initvar_binding) = checker.importer().get_or_import_symbol(
+        &ImportRequest::import("dataclasses", "InitVar"),
+        default.start(),
+        checker.semantic(),
+    )?;
+
+    // Delete the default value. For example,
+    // - def __post_init__(self, foo: int = 0) -> None: ...
+    // + def __post_init__(self, foo: int) -> None: ...
+    let default_edit = Edit::deletion(parameter.end(), default.end());
+
+    // Add `dataclasses.InitVar` field to class body.
+    let locator = checker.locator();
+
+    let content = {
+        let parameter_name = locator.slice(&parameter.name);
+        let default = locator.slice(default);
+        let line_ending = checker.stylist().line_ending().as_str();
+
+        if let Some(annotation) = &parameter
+            .annotation
+            .as_deref()
+            .map(|annotation| locator.slice(annotation))
+        {
+            format!("{parameter_name}: {initvar_binding}[{annotation}] = {default}{line_ending}")
+        } else {
+            format!("{parameter_name}: {initvar_binding} = {default}{line_ending}")
+        }
+    };
+
+    let indentation = indentation_at_offset(post_init_def.start(), checker.locator())
+        .context("Failed to calculate leading indentation of `__post_init__` method")?;
+    let content = textwrap::indent(&content, indentation);
+
+    let initvar_edit = Edit::insertion(
+        content.into_owned(),
+        locator.line_start(post_init_def.start()),
+    );
```

Comment:
Rather than iterate through all statements in the class body until we get the first non-docstring statement, I changed it so that we just insert the `InitVar` symbol in the line before the `__post_init__` definition. It seems much simpler and less error-prone as we already know that we're past the docstring in this case